cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -203,9 +204,8 @@ class DirichletBC
   void dof_values(Eigen::Ref<Eigen::Matrix<T, Eigen::Dynamic, 1>> values) const
   {
     assert(_g);
-    la::VecReadWrapper g(_g->vector().vec());
     for (Eigen::Index i = 0; i < _dofs.rows(); ++i)
-      values[_dofs(i, 0)] = g.x[_dofs(i, 1)];
+      values[_dofs(i, 0)] = _g->x()->array()[_dofs(i, 1)];
```

Comment:
Maybe these should be pulled out as const references before the loop?

Reply:
Yes - change was made quickly.